src/rdkafka_broker.c

```diff
@@ -3995,6 +3995,7 @@ static void rd_kafka_broker_producer_serve(rd_kafka_broker_t *rkb,
         rd_interval_t timeout_scan;
         unsigned int initial_state = rkb->rkb_state;
         rd_ts_t now;
+        rd_ts_t now, overshoot;
```

Comment:
`now` is duplicated.

But we'll want `overshoot` to be defined in the scope where it is used, namely in the while loop below.